Sources/NIOIRC/IRCDispatcher.swift

```diff
@@ -64,7 +64,8 @@ public protocol IRCDispatcher {
                     message    : String) throws
   func doMessage   (sender     : IRCUserID?,
```

Comment:
Would be nice to keep the old API as well for backwards compat, i.e. the one taking no metadata. The default implementation of the one w/ metadata could call into the older method (vs raising notImplemented).